### DISCUSSION TITLE
Welcome to pages Discussions!

### DISCUSSION THREAD

**Discussion by qunabu** [EscolaLMS/pages#1] [General]
Pages Model and API 

generate code with https://www.infyom.com/open-source then refactor 

Please note that this generator is powerful, whole https://github.com/EscolaLMS/Courses was bootstraped with that 

Purpose of this package: Static Pages, eg Terms & Conditions or Privacy Policy, Management of Static Pages, eg Terms & Conditions or Privacy Policy 

Pages model 

* Title 
* Autor (relation to user) 
* Content (HTML/Markdown) 
* Slug (unique) 

Packages should include 

- [ ] migration
- [ ] service provider
- [ ] models
- [ ] CRUD api endpoints
- [ ] auth provider 
- [ ] policy 
- [ ] swagger annotations 
- [ ] seeder 
- [ ] tests

**Comment by qunabu** [comment]
Personaly i do create in the way 

1. create migration file 
2. run migration 
3. create api from table `php artisan infyom:api $MODEL_NAME --fromTable --tableName=$TABLE_NAME`

**Comment by shdpl** [comment]
As far as I'm following the discussion:
It's a CRUD for the 1 model containing four attributes listed above.
My questions are:
* by auth provider you mean a custom provider that gives access to modify *title* and *content* based on the assigned author (user)?
* what seeder should seed the database with? only database schema? if so, should migration just run that seeder? alternatively, do we need some test data?

**Reply by qunabu** [reply]
* Author is user from group `admin` or `tutor` 
* for dummy content you faker https://github.com/fzaninotto/Faker and default Laravels factories https://laravel.com/docs/8.x/database-testing#defining-model-factories

**Reply by qunabu** [reply]
author = someone who have created content, `$this->author_id = Auth::id()` `Auth` is Laravel facade
